packages/bierzo-wallet/src/routes/registerName/index.tsx

```diff
@@ -1,19 +1,66 @@
+import { Identity } from "@iov/bcp";
+import { BnsConnection, ChainAddressPair } from "@iov/bns";
+import { TransactionEncoder } from "@iov/encoding";
+import { FormValues } from "medulas-react-components/lib/components/forms/Form";
+import { ToastContext } from "medulas-react-components/lib/context/ToastProvider";
+import { ToastVariant } from "medulas-react-components/lib/context/ToastProvider/Toast";
 import React from "react";
+import * as ReactRedux from "react-redux";
 
 import { history } from "..";
+import { generateRegisterUsernameTxRequest, sendSignAndPostRequest } from "../../communication/signAndPost";
 import PageMenu from "../../components/PageMenu";
+import { getCodecForChainId } from "../../logic/codec";
+import { getConnectionForChainId } from "../../logic/connection";
+import { RootState } from "../../store/reducers";
 import { BALANCE_ROUTE } from "../paths";
-import Layout from "./components";
-
-async function onSubmit(values: object): Promise<void> {
-  //console.log(values);
-}
+import Layout, { SET_USERNAME_FIELD } from "./components";
 
 function onCancel(): void {
   history.push(BALANCE_ROUTE);
 }
 
 const RegisterUsername = (): JSX.Element => {
+  const toast = React.useContext(ToastContext);
+  const pubKeys = ReactRedux.useSelector((state: RootState) => state.extension.keys);
+  async function onSubmit(values: object): Promise<void> {
+    const formValues = values as FormValues;
+
+    const username = formValues[SET_USERNAME_FIELD];
+
+    let bnsIdentity: Identity | null = null;
+    const addresses: ChainAddressPair[] = [];
```

Comment:
Can you create this `const addresses: ChainAddressPair[]` list only once (when registration component is created) and then pass into AddressesTable as a property? This ensures the UI and the sumitted data is always the same, even if we refactor some of the code at some point in time.

`AddressesTable` can have an `addresses: ChainAddressPair[]` property and then get the chain names on demand as part of the UI logic.

Reply:
Sorting can remain in the `AddressesTable` component, since the order in the transaction does not matter